tests/test_interpolate.py

```diff
@@ -351,3 +352,623 @@ def test_duplicate_idxs_period_pairs():
+    for lamb in [0.0, 0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1.0]:
+        res = pad(interpolate_fn, src_k, dst_k, lamb, lamb_min, lamb_max)
+        if lamb <= lamb_min:
+            np.testing.assert_array_equal(res[0], src_k[0])
+        elif lamb >= lamb_max:
+            np.testing.assert_array_equal(res[0], dst_k[0])
+        else:
+            pass
+
+
+def get_aldehyde():
+    return Chem.MolFromMolBlock(
+        """
+  Mrv2311 11042413482D
+
+  4  3  0  0  0  0            999 V2000
+    4.7768    0.7115    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
+    3.9518    0.7115    0.0000 O   0  0  0  0  0  0  0  0  0  0  0  0
+    5.1893   -0.0030    0.0000 H   0  0  0  0  0  0  0  0  0  0  0  0
+    5.1893    1.4260    0.0000 H   0  0  0  0  0  0  0  0  0  0  0  0
+  1  2  2  0  0  0  0
+  1  3  1  0  0  0  0
+  1  4  1  0  0  0  0
+M  END
+$$$$
+""",
+        removeHs=False,
+    )
+
+
+def get_inv_nitrogen():
+    return Chem.MolFromMolBlock(
+        """
+  Mrv2311 11042413493D
+
+  4  3  0  0  0  0            999 V2000
+   -0.0110   -0.0156    0.0270 N   0  0  0  0  0  0  0  0  0  0  0  0
+    1.0099    0.0080   -0.0138 H   0  0  0  0  0  0  0  0  0  0  0  0
+   -0.3291   -0.4654   -0.8338 H   0  0  0  0  0  0  0  0  0  0  0  0
+   -0.3291    0.9548   -0.0138 H   0  0  0  0  0  0  0  0  0  0  0  0
+  1  3  1  0  0  0  0
+  1  4  1  0  0  0  0
+  1  2  1  0  0  0  0
+M  END
+$$$$""",
+        removeHs=False,
+    )
+
+
+def get_non_inv_nitrogen():
+    return Chem.MolFromMolBlock(
+        """
+  Mrv2311 11042413503D
+
+  4  3  0  0  0  0            999 V2000
+   -0.0138   -0.0196    0.0339 N   0  0  2  0  0  0  0  0  0  0  0  0
+    1.2876    0.0101   -0.0174 F   0  0  0  0  0  0  0  0  0  0  0  0
+   -0.4197   -0.5936   -1.0629 F   0  0  0  0  0  0  0  0  0  0  0  0
+   -0.4197    1.2173   -0.0174 F   0  0  0  0  0  0  0  0  0  0  0  0
+  1  4  1  0  0  0  0
+  1  3  1  0  0  0  0
+  1  2  1  0  0  0  0
+M  END
+$$$$""",
+        removeHs=False,
+    )
+
+
+@pytest.mark.parametrize(
+    "mol_a, mol_b, core",
+    [
+        (get_aldehyde(), get_inv_nitrogen(), np.array([[0, 0], [1, 1], [2, 2], [3, 3]], dtype=np.int32)),
+        (get_inv_nitrogen(), get_aldehyde(), np.array([[0, 0], [1, 1], [2, 2], [3, 3]], dtype=np.int32)),
+    ],
+)
+def test_core_achiral_interpolation_sp2_to_invertible_sp3(mol_a, mol_b, core):
+    # test that we're using the default interpolation schedule spanning over the entirely of lambda [0,1]
+    # for achiral transformations.
+    ff = Forcefield.load_default()
+
+    st = SingleTopology(mol_a, mol_b, core, ff)
+
+    lhs = st.setup_intermediate_state(0.0)
+    rhs = st.setup_intermediate_state(1.0)
+
+    # check that chiral volume is undefined in both end-states
+    assert len(lhs.chiral_atom.potential.idxs) == 0
+    assert len(rhs.chiral_atom.potential.idxs) == 0
+
+    lhs_bond_k = lhs.bond.params[0][0]
```

Comment:
Could this (and several of the tests below) be strengthened to check all interactions (right now seems to only look at the first)? I think you can just vectorize all of the assertions pretty trivially unless I'm missing something. E.g.
```suggestion
    lhs_bond_k = lhs.bond.params[:, 0]
```

Reply:
re-used the `_assert_bonded_term` for this.